src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -1032,4 +1053,144 @@ private StringConcatFactory() {
+        static final MethodTypeDesc APPEND_BOOLEAN_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_boolean);
+        static final MethodTypeDesc APPEND_CHAR_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_char);
+        static final MethodTypeDesc APPEND_DOUBLE_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_double);
+        static final MethodTypeDesc APPEND_FLOAT_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_float);
+        static final MethodTypeDesc APPEND_INT_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_int);
+        static final MethodTypeDesc APPEND_LONG_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_long);
+        static final MethodTypeDesc APPEND_OBJECT_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_Object);
+        static final MethodTypeDesc APPEND_STRING_TYPE = MethodTypeDesc.of(STRING_BUILDER, ConstantDescs.CD_String);
+        static final MethodTypeDesc INT_VOID_TYPE = MethodTypeDesc.of(ConstantDescs.CD_void, ConstantDescs.CD_int);
+
+        /**
+         * Ensure a capacity in the initial StringBuilder to accommodate all constants plus this factor times the number
+         * of arguments.
+         */
+        static final int ARGUMENT_SIZE_FACTOR = 4;
+
+        public static final Set<Lookup.ClassOption> SET_OF_STRONG = Set.of(STRONG);
+
+        static {
+            DUMPER = ClassFileDumper.getInstance("java.lang.invoke.StringConcatFactory.dump", "stringConcatClasses");
+        }
+
+        private SimpleStringBuilderStrategy() {
+            // no instantiation
+        }
+
+        private static MethodHandle generate(Lookup lookup, MethodType args, String[] constants) throws Exception {
+            String className = getClassName(lookup.lookupClass());
+
+            byte[] classBytes = ClassFile.of().build(ClassDesc.of(className),
+                    new Consumer<ClassBuilder>() {
+                        @Override
+                        public void accept(ClassBuilder clb) {
+                            clb.withFlags(AccessFlag.FINAL, AccessFlag.SUPER, AccessFlag.SYNTHETIC)
+                                .withMethodBody(METHOD_NAME,
+                                        MethodTypeDesc.ofDescriptor(args.toMethodDescriptorString()),
+                                        ClassFile.ACC_FINAL | ClassFile.ACC_PRIVATE | ClassFile.ACC_STATIC,
+                                        generateMethod(constants, args));
+                    }});
+
+            try {
+                Lookup hiddenLookup = lookup.makeHiddenClassDefiner(className, classBytes, SET_OF_STRONG, DUMPER)
+                                            .defineClassAsLookup(true);
+                Class<?> innerClass = hiddenLookup.lookupClass();
+                return hiddenLookup.findStatic(innerClass, METHOD_NAME, args);
+            } catch (Exception e) {
+                DUMPER.dumpFailedClass(className, classBytes);
```

Comment:
This line is no longer needed.   The bytes will be dumped if it's enabled for both success and failing case.